src/platform/android/DiagnosticDataProviderImpl.cpp

```diff
@@ -37,5 +45,171 @@ DiagnosticDataProviderImpl & DiagnosticDataProviderImpl::GetDefaultInstance()
     return sInstance;
 }
 
+void DiagnosticDataProviderImpl::InitializeWithObject(jobject manager)
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrReturn(env != nullptr,
+                   ChipLogError(DeviceLayer, "Failed to GetEnvForCurrentThread for DiagnosticDataProviderManagerImpl"));
+
+    mDiagnosticDataProviderManagerObject = env->NewGlobalRef(manager);
+    VerifyOrReturn(mDiagnosticDataProviderManagerObject != nullptr,
+                   ChipLogError(DeviceLayer, "Failed to NewGlobalRef DiagnosticDataProviderManager"));
+
+    jclass DiagnosticDataProviderManagerClass = env->GetObjectClass(manager);
+    VerifyOrReturn(DiagnosticDataProviderManagerClass != nullptr,
+                   ChipLogError(DeviceLayer, "Failed to get DiagnosticDataProviderManager Java class"));
+
+    mGetRebootCountMethod = env->GetMethodID(DiagnosticDataProviderManagerClass, "getRebootCount", "()I");
+    if (mGetRebootCountMethod == nullptr)
+    {
+        ChipLogError(DeviceLayer, "Failed to access DiagnosticDataProviderManager 'getRebootCount' method");
+        env->ExceptionClear();
+    }
+
+    mGetNifMethod =
+        env->GetMethodID(DiagnosticDataProviderManagerClass, "getNetworkInterfaces", "()[Lchip/platform/NetworkInterface;");
+    if (mGetNifMethod == nullptr)
+    {
+        ChipLogError(DeviceLayer, "Failed to access DiagnosticDataProviderManager 'getNetworkInterfaces' method");
+        env->ExceptionClear();
+    }
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
+{
+    chip::DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrExit(mDiagnosticDataProviderManagerObject != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mGetRebootCountMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != NULL, err = CHIP_JNI_ERROR_NO_ENV);
+    ChipLogProgress(DeviceLayer, "Received GetRebootCount");
+
+    {
+        jint count = env->CallIntMethod(mDiagnosticDataProviderManagerObject, mGetRebootCountMethod);
+        if (count > UINT16_MAX)
+        {
+            return CHIP_ERROR_INVALID_INTEGER_VALUE;
+        }
+        rebootCount = static_cast<uint16_t>(count);
+    }
+
+exit:
```

Comment:
I think we should prefer to use VerifyOrReturn(), same elsewhere

Reply:
fixed